ods_ci/tests/Resources/Files/pipeline-samples/ray_integration.py

```diff
@@ -0,0 +1,86 @@
+from kfp import components, dsl
+from ods_ci.libs.DataSciencePipelinesKfpTekton import DataSciencePipelinesKfpTekton
+
+
+def ray_fn(openshift_server:str, openshift_token:str) -> int:
+    from codeflare_sdk.cluster.cluster import Cluster, ClusterConfiguration
+    from codeflare_sdk.cluster.auth import TokenAuthentication
+    import ray
+
+
+    auth = TokenAuthentication(
+        token=openshift_token,
+        server=openshift_server,
+        skip_tls=True
+    )
+    auth.login()
+    cluster = Cluster(ClusterConfiguration(
+        name='raytest',
+        namespace='dlovison-ray',
+        num_workers=1,
+        min_cpus=1,
+        max_cpus=1,
+        min_memory=1,
+        max_memory=1,
+        num_gpus=0,
+        image="quay.io/project-codeflare/ray:latest-py39-cu118",
+        instascale=False
+    ))
+    # workaround for https://github.com/project-codeflare/codeflare-sdk/pull/412
+    cluster_file_name = '/opt/app-root/src/.codeflare/appwrapper/raytest.yaml'
+    # Read in the file
+    with open(cluster_file_name, 'r') as file:
+        filedata = file.read()
+
+    # Replace the target string
+    filedata = filedata.replace('busybox:1.28', 'quay.io/project-codeflare/busybox:latest')
+
+    # Write the file out again
+    with open(cluster_file_name, 'w') as file:
+        file.write(filedata)
+    # end workaround
+
+    # always clean the resources
+    cluster.down()
+    print(cluster.status())
+    cluster.up()
+    cluster.wait_ready()
+    print(cluster.status())
+    print(cluster.details())
+
+    ray_dashboard_uri = cluster.cluster_dashboard_uri()
+    ray_cluster_uri = cluster.cluster_uri()
+    print(ray_dashboard_uri)
+    print(ray_cluster_uri)
+
+    # before proceeding make sure the cluster exists and the uri is not empty
+    assert ray_cluster_uri, "Ray cluster needs to be started and set before proceeding"
+
+    # reset the ray context in case there's already one.
+    ray.shutdown()
+    # establish connection to ray cluster
+    ray.init(address=ray_cluster_uri)
+    print("Ray cluster is up and running: ", ray.is_initialized())
+
+    @ray.remote
+    def train_fn():
+        return 100
```

Comment:
Thinking whether to use some "real" function doing some training, instead of this stub.

Reply:
This is the basic test. We can create more complex tests in a different test.